### ISSUE TITLE
Wandb Validation Loss logs a strange value at Last_epoch + 1

### ISSUE THREAD

**Issue by thomassajot** [ultralytics/yolov5#9646]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Validation, Other

### Bug
Training a model using: (red line / green line) 
```bash
python train.py --weights yolov5x6.pt --data config.yaml --hyp hyps.yaml --img 1280 --epoch 100 --project runs/traffic_light_2020_undistorted --name yolov5x6_1280_cross_prod_smoothing_0.05 --batch-size 16 --rect --label-smoothing 0.05
```

```bash
train.py --weights yolov5x6.pt --data config.yaml --hyp hyps.yaml --img 1280 --epoch 100 --project runs/traffic_light_2020_undistorted --name yolov5x6_1280_cross_prod --batch-size 16 --rect
```

At the end of training, there is a validation at epoch 100 which has a very suspicious value (see image) 
1. Is it normal to have a validation at epoch 100 ? (since the epoch count starts at 0, I would expect the last validation to be assigned to epoch 99) 
2. Is the model evaluated in a different way for this last step ? 

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/15252203/193008402-88337775-5cdc-4a08-aaf7-3cb022e71237.png">

### Environment
- YOLOv5x6 torch 1.12 + CUDA:11.2 (V100) 
- Python 3.9.13
- Ubuntu 18.04.6

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by soumik12345** [comment]
Hi @thomassajot 

>Is it normal to have a validation at epoch 100 ? (since the epoch count starts at 0, I would expect the last validation to be assigned to epoch 99)

While logging on WandB, the 0th step is basically the initial model step prior to training, and the epoch count is logged from 1.

> Is the model evaluated in a different way for this last step ?

No, all steps are evaluated the same. The sudden drop in loss could be due to the learning rate scheduling in YOLO. Is it possible for you to share a minimal reproducible example or the WandB workspace? That could give us a clue as to what could the cause of such an abrupt drop.

**Comment by thomassajot** [comment]
Unfortunately, I cannot share the code to reproduce this example. The loss value does not seem to have changed for this last epoch. (see picture).
<img width="1422" alt="image" src="https://user-images.githubusercontent.com/15252203/193077165-3e0738f2-cd7c-471c-a030-9ba5964783a4.png">

I will do an experiment with 150 epochs and see if the drop is consistent.

**Comment by thomassajot** [comment]
I believe the last evaluation is performed on the selected `best` weights. Has shown in the validation plots, the loss is getting worst after epoch ~20. I believe the best weights are taken from around there (using best MAP@50). So the last evaluation will replicate the loss performance at a previous epoch.
